pkg/controllers/csi/metadata/sqlite_gorm_client_test.go

```diff
@@ -0,0 +1,392 @@
+	require.NoError(t, err)
+
+	var result []OSMount
+
+	db.db.Raw(`SELECT * FROM os_mounts WHERE tenant_uuid = "abc123"`).Scan(&result)
+	assert.NotEmpty(t, result)
+	assert.True(t, result[0].DeletedAt.Valid)
+}
+
+func TestCreateAppMount(t *testing.T) {
+	db, err := setupSqLiteDB(PostReconcile)
+	require.NoError(t, err)
+
+	tenantConfig, err := db.ReadTenantConfigByTenantUUID(context.Background(), "abc123")
+	require.NoError(t, err)
+
+	cm, err := db.ReadCodeModuleByVersion(context.Background(), tenantConfig.DownloadedCodeModuleVersion)
+	require.NoError(t, err)
+
+	appMount := &AppMount{
+		VolumeMeta: VolumeMeta{
+			ID:                "appmount1",
+			PodUid:            "pod111",
+			PodName:           "podiv",
+			PodNamespace:      "testnamespace",
+			PodServiceAccount: "podsa",
+		},
+		Location:      "loc1",
+		MountAttempts: 0,
+		CodeModule:    *cm,
+	}
+
+	err = db.CreateAppMount(context.Background(), appMount)
+	require.NoError(t, err)
+
+	var result []AppMount
+
+	db.db.Raw(`SELECT * FROM app_mounts WHERE volume_meta_id = "appmount1"`).Scan(&result)
+	assert.NotEmpty(t, result)
+}
+
+func TestReadAppMounts(t *testing.T) {
+	db, err := setupSqLiteDB(PostPublish)
+	require.NoError(t, err)
+
+	appMounts, err := db.ReadAppMounts(context.Background())
+	require.NoError(t, err)
+
+	assert.NotNil(t, appMounts)
+	assert.NotEmpty(t, len(appMounts))
+	assert.Equal(t, "appmount1", appMounts[0].VolumeMeta.ID)
+}
+
+func TestReadAppMount(t *testing.T) {
+	db, err := setupSqLiteDB(PostPublish)
+	require.NoError(t, err)
+
+	appMount, err := db.ReadAppMountByVolumeMetaID(context.Background(), "appmount1")
+	require.NoError(t, err)
+
+	assert.NotNil(t, appMount)
+	assert.Equal(t, "appmount1", appMount.VolumeMeta.ID)
+
+	_, err = db.ReadAppMountByVolumeMetaID(context.Background(), "")
+	require.Error(t, err)
+
+	_, err = db.ReadAppMountByVolumeMetaID(context.Background(), "unknown")
+	require.Error(t, err)
+}
+
+func TestUpdateAppMount(t *testing.T) {
+	db, err := setupSqLiteDB(PostPublish)
+	require.NoError(t, err)
+
+	appMount, err := db.ReadAppMountByVolumeMetaID(context.Background(), "appmount1")
+	require.NoError(t, err)
+
+	appMount.MountAttempts = 5
+
+	err = db.UpdateAppMount(context.Background(), appMount)
+	require.NoError(t, err)
+
+	var result []OSMount
+
+	db.db.Raw(`SELECT * FROM app_mounts WHERE volume_meta_id = "appmount1"`).Scan(&result)
+	assert.NotEmpty(t, result)
+	assert.Equal(t, int64(5), result[0].MountAttempts)
+}
+
+func TestSoftDeleteAppMount(t *testing.T) {
+	db, err := setupSqLiteDB(PostPublish)
+	require.NoError(t, err)
+
+	appMount, err := db.ReadAppMountByVolumeMetaID(context.Background(), "appmount1")
+	require.NoError(t, err)
+
+	assert.NotNil(t, appMount)
+	assert.Equal(t, "appmount1", appMount.VolumeMeta.ID)
+
+	err = db.DeleteAppMount(context.Background(), appMount)
+	require.NoError(t, err)
+
+	var result []AppMount
+
+	db.db.Raw(`SELECT * FROM app_mounts WHERE volume_meta_id = "appmount1"`).Scan(&result)
```

Comment:
why not using GORM? The purpose to migrate to GORM to avoid RAW sqls :)

Reply:
No need to test ORM, orm works it has it's own unit tests. Focus on testing our stuff. The whole purpose of rewriting is to remove raw SQL everywhere.